### DISCUSSION TITLE
Paginated Queries vs Infinite Queries

### DISCUSSION THREAD

**Discussion by bestickley** [TanStack/query#4372] [Q&A] [answered]
Hi, I've implemented a paginated table with `useInfiniteQuery` but then I stumbled upon the [Paginated Queries](https://tanstack.com/query/v4/docs/guides/paginated-queries) section of the documentation. Which is recommended for paging through a table?

**Comment by TkDodo** [comment, marked as answer]
I think both can work, but paginated queries usually work better with client side pagination while infinite queries work better when the server controls what the next page is.

**Reply by bestickley** [reply]
client-side vs server-side pagination distinction helped a lot. Thank you!
